packages/graphile-build-pg/src/plugins/PgTypesPlugin.js

```diff
@@ -600,21 +604,10 @@ export default (function PgTypesPlugin(
       pg2GqlMapper[1186] = {
         map: str => parseInterval(str),
         unmap: o => {
-          const keys = [
-            "seconds",
-            "minutes",
-            "hours",
-            "days",
-            "months",
-            "years",
-          ];
-          const parts = [];
-          for (const key of keys) {
-            if (o[key]) {
-              parts.push(`${o[key]} ${key}`);
-            }
-          }
-          return sql.value(parts.join(" ") || "0 seconds");
+          const interval = Object.assign(rawParseInterval(), o);
+          return sql.fragment`pg_catalog.justify_interval(${sql.value(
+            interval.toPostgres()
+          )})`;
```

Comment:
Please don't use justify_interval; it changes the meaning of the interval. Note that the first is correct, but the second (using justify_interval) is incorrectly 2 days later!

```
[test] # select '2019-02-01T00:00:00Z'::timestamptz + interval '1 month -1 hour';
┌────────────────────────┐
│        ?column?        │
├────────────────────────┤
│ 2019-02-28 23:00:00+00 │
└────────────────────────┘
(1 row)

Time: 0.216 ms
[test] # select '2019-02-01T00:00:00Z'::timestamptz + justify_interval(interval '1 month -1 hour');
┌────────────────────────┐
│        ?column?        │
├────────────────────────┤
│ 2019-03-02 23:00:00+00 │
└────────────────────────┘
(1 row)

Time: 7.468 ms
```

Even without the negative values you can see the results differ:

```
[test] # select '2019-02-01T00:00:00Z'::timestamptz + interval '30 days';
┌────────────────────────┐
│        ?column?        │
├────────────────────────┤
│ 2019-03-03 00:00:00+00 │
└────────────────────────┘
(1 row)

Time: 0.234 ms
[test] # select '2019-02-01T00:00:00Z'::timestamptz + justify_interval(interval '30 days');
┌────────────────────────┐
│        ?column?        │
├────────────────────────┤
│ 2019-03-01 00:00:00+00 │
└────────────────────────┘
(1 row)

Time: 0.245 ms
```